tests/integration/modules/test_win_pkg.py

```diff
@@ -33,8 +34,24 @@ def test_adding_removing_pkg_sls(self):
         Test add and removing a new pkg sls
         in the windows software repository
         '''
-        def _check_pkg(pkgs, exists=True):
-            self.run_function('pkg.refresh_db')
+        def _check_pkg(pkgs, exists=True, check_refresh=None):
+            refresh = self.run_function('pkg.refresh_db')
+            if check_refresh:
+                count = 2
+                while count != 0:
+                    try:
+                        self.assertEqual(0, refresh['failed'],
+                             msg='failed returned {0}. Expected return: 0'.format(refresh['failed']))
+                        self.assertEqual(check_refresh, refresh['total'],
+                             msg='total returned {0}. Expected return {1}'.format(check_refresh, refresh['total']))
+                        self.assertEqual(check_refresh, refresh['success'],
+                             msg='success returned {0}. Expected return {1}'.format(check_refresh, refresh['success']))
+                        count = 0
+                    except AssertionError as err:
+                        if count == 1:
+                            raise AssertionError(err)
+                        count = count -1
+                        refresh = self.run_function('pkg.refresh_db')
             repo_data = self.run_function('pkg.get_repo_data', timeout=300)
```

Comment:
timeout=30   it just opens an existing file and reads it into ram.  30 seconds is heaps

Reply:
looks like that change was made intentionally to also try to stabilize this flaky test so I dont want to remove it quit yet.

Reply:
You could also call pkg.refresh_db with verbose=True and failhard=False
The data structure return will be this even if it fails (with data not like the example below). 
```
{
    "local": {
        "total": 0,
        "success": 0,
        "success_list": {},
        "failed_list": {},
        "failed": 0
    }
}
```